### PR TITLE
Flow-typed support in Gestalt for third party packages

### PR DESCRIPTION
**Pull request by pfarejowicz**
This change adds support for [Flow-typed](https://github.com/flowtype/flow-typed) to declare Flow definitions for our third party dependencies. I manually ran the install command on a few of our most used packages, but this is open to adding more in the future. This allows us to pull Flow definitions from the open-source community and keep up to date with upgrades.

I disabled eslint and prettier on the new `flow-typed` folder because those are auto-generated so we should not bother touching them.

### CODE REVIEW

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit 9afa5a84fd5ed31dd0cb93fe658585ec8d39cbaa

https://deploy-preview-174--gestalt.netlify.com

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please add a [changelog.md](https://github.com/pinterest/gestalt/blob/master/CHANGELOG.md) entry for your changes.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/gestalt/src/Box/Box.js [hunk 1]
```diff
@@ -744,32 +744,32 @@ const PaddingPropType = PropTypes.oneOf([
 
 Box.propTypes = {
   children: PropTypes.node,
-  dangerouslySetInlineStyle: PropTypes.exact({
+  dangerouslySetInlineStyle: PropTypes.shape({
```
Comment (chrislloyd) [inline packages/gestalt/src/Box/Box.js]: Hey what was your reasoning for this change? Seems to make the interface more loose.

Comment (pfarejowicz) [reply packages/gestalt/src/Box/Box.js]: @chrislloyd Flow isnt happy with that one (see https://github.com/pfarejowicz/gestalt/blob/8967902e3e96725c1c3113a8caf58acf7d566978/flow-typed/npm/prop-types_v15.x.x.js ). I see `PropTypes.exact` in the source code, but Facebook does not list it on the docs site (not sure why) (see https://github.com/facebook/prop-types ) so I think there is some miscommunication and a bug in flow-typed

Comment (chrislloyd) [reply packages/gestalt/src/Box/Box.js]: I don't think changing our code here is the right solution to get flow safety here. Are there other approaches?

**Pull request merged**